rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java

```diff
@@ -749,42 +749,67 @@ public JobId submit(URL job, Map<String, String> variables, Map<String, String>
+    public JobId submit(URL job, Map<String, String> variables)
             throws NotConnectedException, PermissionException, SubmissionClosedException, JobCreationException {
-        return this.submitFromCatalog(catalogRestURL,
-                                      bucketName,
-                                      workflowName,
-                                      Collections.<String, String> emptyMap());
+        return this.submit(job, variables, null);
+    }
 
+    @Override
+    public JobId submit(Map<String, String> genericInfos, URL job, Map<String, String> variables)
+            throws NotConnectedException, PermissionException, SubmissionClosedException, JobCreationException {
+        return this.submit(job, variables, genericInfos, null);
     }
 
     @Override
-    public JobId submit(File job, Map<String, String> variables)
+    public JobId submit(URL job, Map<String, String> variables, Map<String, String> requestHeaderParams)
+            throws NotConnectedException, PermissionException, SubmissionClosedException, JobCreationException {
+        return submit(job, variables, null, requestHeaderParams);
+    }
+
+    @Override
+    public JobId submit(URL job, Map<String, String> variables, Map<String, String> genericInfos,
+            Map<String, String> requestHeaderParams)
             throws NotConnectedException, PermissionException, SubmissionClosedException, JobCreationException {
         JobIdData jobIdData = null;
-        try (InputStream is = new FileInputStream(job)) {
-            jobIdData = restApiClient().submitXml(sid, is, variables);
+        try {
+            URLConnection urlConnection = job.openConnection();
+
+            for (Map.Entry<String, String> requestHeaderEntry : requestHeaderParams.entrySet()) {
+                urlConnection.addRequestProperty(requestHeaderEntry.getKey(), requestHeaderEntry.getValue());
+            }
+            InputStream is = urlConnection.getInputStream();
+            jobIdData = restApiClient().submitXml(sid, is, variables, genericInfos);
         } catch (Exception e) {
             throwNCEOrPEOrSCEOrJCE(e);
         }
         return jobId(jobIdData);
     }
 
     @Override
-    public JobId submit(URL job, Map<String, String> variables)
+    public JobId submitFromCatalog(String catalogRestURL, String bucketName, String workflowName)
             throws NotConnectedException, PermissionException, SubmissionClosedException, JobCreationException {
-        return this.submit(job, variables, Collections.<String, String> emptyMap());
+        return this.submitFromCatalog(catalogRestURL, bucketName, workflowName, null);
+
     }
 
     @Override
     public JobId submitFromCatalog(String catalogRestURL, String bucketName, String workflowName,
             Map<String, String> variables)
             throws NotConnectedException, PermissionException, SubmissionClosedException, JobCreationException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)